source/Ship.cpp

```diff
@@ -3760,6 +3760,230 @@ void Ship::AddOutfit(const Outfit *outfit, int count)
 
 
 
+// Add or reduce an attribute. (To reduce, pass a negative number.)
+void Ship::ChangeAttribute(string targetAttribute, double modifyAmount)
+{
+	if(!targetAttribute.empty())
+	{
+		double limiter = 0.;
+		double minAttributeValue = 0.;
+		double originalBaseValue = baseAttributes.Get(targetAttribute);
+		double originalValue = attributes.Get(targetAttribute);
+		double newBaseValue = originalBaseValue + modifyAmount;
+		double newValue = originalValue + modifyAmount;
+
+		// Safety checks to ensure the new value is within parameters.
+		if(newBaseValue < 1. && newValue < 1. && targetAttribute == "hull")
+		{
+			// This means the minimum value for this attribute is 1.0
+			minAttributeValue = 1.;
+			limiter = minAttributeValue - newBaseValue;
+		}
+		else if(newBaseValue < 0.01 && (targetAttribute == "drag" || targetAttribute == "mass"))
+		{
+			// This means the minimum value for these attributes is 0.01
+			minAttributeValue = 0.01;
+			limiter = minAttributeValue - newBaseValue;
+		}
+		// Special handling for attributes that cannot be less than 0.
+		// These values can be 0, just they cannot be negative.
+		else if(newBaseValue < 0. && (targetAttribute == "outfit space" || targetAttribute == "cargo space" ||
```

Comment:
Been a while I didnt code
Can we make a list of them and check if it's in the list or smth, I think it'd be cleaner? (And declare it in the namespace above to use it below too)

Reply:
I'm not actually sure how to do that...